meeting.js

```diff
@@ -0,0 +1,330 @@
+    }
+    microphoneButton.innerHTML = '<i class="fas fa-microphone-slash icon red"></i>';
+    microphoneButton.style.padding = "14px 17px";
+    isMuted = false;
+  }
+}
+
+// 16) Added the Local Share System using Share Button.
+shareButton.addEventListener('click', localshare);
+async function localshare() {
+  const [localShareVideoStream, localShareAudioStream] =
+    await webex.meetings.mediaHelpers.createDisplayStreamWithAudio();
+  await createdMeeting.publishStreams({
+    screenShare: {
+      video: localShareVideoStream,
+      audio: localShareAudioStream,
+    }
+  });
+}
+
+// 17) Enabled BNR using Event Listeners and toggleBNR Function.
+bnrButton.addEventListener('click', toggleBNR);
+async function toggleBNR() {
+  let bnrEffect = null;
+  if (!localStream || !localStream.microphone) {
+    console.error('No local microphone stream available.');
+    return;
+  }
+
+  try {
+    if (!bnrEffect) {
+      const audioContext = new AudioContext({ sampleRate: 48000 });
+      bnrEffect = await webex.meetings.createNoiseReductionEffect(audioContext);
+      await localStream.microphone.addEffect(bnrEffect);
+    }
+
+    if (bnrEffect.isEnabled) {
+      await bnrEffect.disable();
+      bnrButton.classList.remove('selected');
+      console.log('BNR disabled');
+    } else {
+      await bnrEffect.enable();
+      console.log('BNR enabled');
+      bnrButton.classList.add('selected');
+    }
+  } catch (error) {
+    console.error('Error toggling BNR:', error);
+  }
+}
+
+// 18) Enabled VBG using Event Listeners and toggleVBG Function.
+vbgButton.addEventListener('click', toggleVBG);
+async function toggleVBG() {
+  if (!localStream || !localStream.camera || !localStream.camera.outputStream) {
+    console.error('No local video stream available.');
+    return;
+  }
+  try {
+    if (!vbgEffect) {
+      vbgEffect = await webex.meetings.createVirtualBackgroundEffect();
+    }
+    if (vbgEffect.isEnabled) {
+      await vbgEffect.disable();
+      vbgButton.classList.remove('selected');
+    } else {
+      await localStream.camera.addEffect(vbgEffect);
+      await vbgEffect.enable();
+      console.log('VBG enabled');
+      vbgButton.classList.add('selected');
+    }
+  } catch (error) {
+    console.error('Error toggling VBG:', error);
+  }
+}
+
+// 19) Permission Handling
+async function handlePermissionChange(permissionName) {
+  const permission = await navigator.permissions.query({ name: permissionName });
+  permission.onchange = async () => {
+    if (permission.state !== 'granted') {
+      if (permissionName === 'camera') {
+        if (isVideoStarted) {
+          toggleVideo();
+          vbgEffect=null;
+        } 
+        
+      } else if (permissionName === 'microphone') {
+        if (isMuted) {
+          await toggleMicrophone();
+        }
+      }
+    }
+  }
+}
+// Monitor permission changes
+handlePermissionChange('camera');
+handlePermissionChange('microphone');
```

Comment:
What is the purpose of handling permission change here?

Reply:
When the Camera Permission and microphone permission was off it will automatically off the local video and microphone so we can on the microphone and camera instead of another button called load camera

Reply:
Ideally, start video would load camera from inside. In the Kitchen Sink app, we have a separate button to indicate there's a separate API. 

If the permission changes, the page needs to be reloaded anyway, right?

Reply:
Like It will show reloaded option but what if the user in the meeting middle so I had add these permission also